lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -205,39 +212,74 @@ SessionState::SessionState(
     const SessionConfig& cfg, StaticRuleStore& rule_store)
     : imsi_(imsi),
       session_id_(session_ctx_id),
-      local_teid_(0),
       // Request number set to 1, because request 0 is INIT call
       request_number_(1),
       /*current state would be CREATING and version would be 0 */
       curr_state_(CREATING),
       config_(cfg),
       current_version_(0),
+      rtx_counter_(0),
       static_rules_(rule_store) {}
 
 /* get-set methods of new messages  for 5G*/
 uint32_t SessionState::get_current_version() {
   return current_version_;
 }
 
+/* method to set update the session current version */
 void SessionState::set_current_version(
-    int new_session_version, SessionStateUpdateCriteria& session_uc) {
+    uint32_t new_session_version, SessionStateUpdateCriteria& session_uc) {
   current_version_                      = new_session_version;
   session_uc.is_current_version_updated = true;
   session_uc.updated_current_version    = new_session_version;
-  MLOG(MINFO) << " Current version is " << get_current_version();
+  MLOG(MDEBUG) << " Current version is " << get_current_version();
 }
+
 /* Add PDR rule to this rules session list */
-void SessionState::insert_pdr(SetGroupPDR* rule) {
-  PdrList_.push_back(*rule);
+void SessionState::insert_pdr(
+    SetGroupPDR* rule, SessionStateUpdateCriteria& session_uc, bool crit_add) {
+  // Check if it already exists
+  int32_t Pdr_index;
```

Comment:
Pdr_index -> pdr_index

Reply:
We have added unit tests for all pipelined routines ?  few more we will add going forward.